app/src/main/java/com/bps/gotwinter2021/ui/favorites/FavoritesAdapter.kt

```diff
@@ -0,0 +1,50 @@
+package com.bps.gotwinter2021.ui.favorites
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.bps.gotwinter2021.data.model.GOTResponse
+import com.bps.gotwinter2021.databinding.FavoritesItemCardViewBinding
+import com.bps.gotwinter2021.favorites.database.Favorite
+
+class FavoritesAdapter(private val onClickListener: FavoritesClickListener) : ListAdapter<Favorite, FavoritesAdapter.FavoritesViewHolder>(DiffCallback){
+    class FavoritesViewHolder(private var binding: FavoritesItemCardViewBinding): RecyclerView.ViewHolder(binding.root) {
+        fun bind(favorite: Favorite, clickListener: FavoritesClickListener){
+            binding.favorite = favorite
+            binding.favCardItemViewNameTV.text = favorite.characterName
+            binding.favCardItemViewFavoriteIV.setOnClickListener{
+                clickListener.onFavoriteClick(favorite, "favorite")
+            }
+            binding.favCardItemViewCharacterIV.setOnClickListener{
+                clickListener.onNavClick(favorite, "navigate")
+            }
+            binding.executePendingBindings()
+        }
+    }
+
+    companion object DiffCallback: DiffUtil.ItemCallback<Favorite>(){
+        override fun areItemsTheSame(oldItem: Favorite, newItem: Favorite): Boolean {
+            return oldItem === newItem
```

Comment:
i think you have to compare the name here, 
oldItem.characterName == newItem.characterName

Reply:
you're right thanks

Reply:
wait are you sure? I have _oldItem.characterName == newItem.characterName_ in the areContetentsTheSame function.

Reply:
Yes, because in areItemsSame you have to compare if it is the same Item, and you compare it with the id, or in this case with the name, 
and in areContentsTheSame you compare all the item(oldItem) with the newItem
![diffutil](https://user-images.githubusercontent.com/70727528/141143301-f6355c43-8d42-4320-8a69-837f9447ca10.jpg)

Reply:
I got it from the udacity course
![===](https://user-images.githubusercontent.com/91076848/141145251-39728589-9050-49a6-beb9-affd796b2990.png)

Reply:
![image](https://user-images.githubusercontent.com/70727528/141146406-d9948376-1885-4a3c-9001-31917229ce12.png)

Reply:
this is from the Udacity course too